src/main/java/org/craftercms/studio/impl/v2/service/security/internal/EncryptionServiceInternalImpl.java

```diff
@@ -19,51 +19,90 @@
+    private final StudioConfiguration studioConfiguration;
+    private final AuditServiceInternal auditServiceInternal;
+    private final TextEncryptor textEncryptor;
+    private final SiteFeedMapper siteFeedMapper;
+    private final int maxLength;
+    private final long delay;
 
-    protected TextEncryptor textEncryptor;
+    @ConstructorProperties({"studioConfiguration", "auditServiceInternal", "textEncryptor", "siteFeedMapper",
+            "maxLength", "delay"})
+    public EncryptionServiceInternalImpl(StudioConfiguration studioConfiguration, AuditServiceInternal auditServiceInternal,
+                                         TextEncryptor textEncryptor, SiteFeedMapper siteFeedMapper, int maxLength, long delay) {
+        this.studioConfiguration = studioConfiguration;
+        this.auditServiceInternal = auditServiceInternal;
+        this.textEncryptor = textEncryptor;
+        this.siteFeedMapper = siteFeedMapper;
+        this.maxLength = maxLength;
+        this.delay = delay;
+    }
 
     @Override
-    public String encrypt(final String text) throws ServiceLayerException {
+    public String encrypt(final String siteId, final String text) throws ServiceLayerException {
         if (StringUtils.isEmpty(text) || text.length() > maxLength) {
             throw new InvalidParametersException("The provided text is invalid");
         }
         try {
             Thread.sleep(delay * 1000);
-            return textEncryptor.encrypt(text);
+            String encryptedToken = textEncryptor.encrypt(text);
+
+            var auth = SecurityContextHolder.getContext().getAuthentication();
+            String resolvedSiteId = StringUtils.isNotEmpty(siteId) ? siteId
+                    : studioConfiguration.getProperty(CONFIGURATION_GLOBAL_SYSTEM_SITE);
+            String targetId = UUID.randomUUID().toString();
+            createEncryptionAuditLog(resolvedSiteId, auth.getName(), targetId);
+            logger.debug("Encryption token created for site '{}', target ID '{}'", resolvedSiteId, targetId);
+
+            return encryptedToken;
         } catch (CryptoException | InterruptedException e) {
             throw new ServiceLayerException("Error encrypting text", e);
         }
     }
 
-    public long getDelay() {
-        return delay;
-    }
-
-    public void setDelay(long delay) {
-        this.delay = delay;
-    }
-
-    public int getMaxLength() {
-        return maxLength;
-    }
-
-    public void setMaxLength(int maxLength) {
-        this.maxLength = maxLength;
-    }
-
-    public void setTextEncryptor(TextEncryptor textEncryptor) {
-        this.textEncryptor = textEncryptor;
+    /**
+     * Create encryption audit log
+     * @param siteId site identifier
+     * @param actor actor
+     * @param targetId encryption target id
+     */
+    private void createEncryptionAuditLog(String siteId, String actor, String targetId) {
+        SiteFeed site = siteFeedMapper.getSite(Map.of(SITE_ID, siteId));
```

Comment:
Let's use SiteDAO instead of SiteFeedMapper.

Reply:
Updated to use SiteDAO